engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1130,6 +1138,128 @@ namespace dmGameSystem
+     *     assert( grp == hash("enemy") )
+     * end
+     * ```
+     */    
+    static int Physics_GetGroup(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        
+        dmhash_t group_hash = dmGameSystem::GetCollisionGroup(comp_world, comp);
+        dmScript::PushHash(L, group_hash);
+        
+        return 1;
+    }    
+
+    /*# updates the mask of a collision object
+     *
+     * Sets or clears the masking of a group (maskbit) in a collision object.
+     *
+     * @name physics.set_maskbit
+     * @param url [type:string|hash|url] the collision object to change the mask of.
+     * @param group [type:string] the name of the group (maskbit) to modify in the mask.
+     * @param [type:boolean] boolean value of the new maskbit. 'true' to enable, 'false' to disable.
+     * ```lua
+     * function example()
+     *     -- no longer collide with the "enemy" group
+     *     physics.set_maskbit("#collisionobject","user",enemy)
+     * end
+     * ```
+     */
+    static int Physics_SetMaskBit(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        bool boolvalue = CheckBoolean(L, 3);
+        
+        dmGameSystem::SetCollisionMaskBit(comp_world, comp, group_id, boolvalue);
+
+        return 0;
+    }    
+    
+    /*# checks the presense of a group in the mask (maskbit) of a collision object
+     *
+     * Returns true if the specified group is set in the mask of a collision 
+     * object, false otherwise.
+     *
+     * @name physics.get_maskbit
+     * @param url [type:string|hash|url] the collision object to check the mask of.
+     * @param group [type:string] the name of the group to check for.
+     * @return [type:boolean] boolean value of the maskbit. 'true' if present, 'false' otherwise.
+     * ```lua
+     * function example()
+     *     -- to check if the collisionobject would collide with "user" group
+     *     local hits_user = physics.get_maskbit("#collisionobject","user")
+     * end
+     * ```
+     */    
+    static int Physics_GetMaskBit(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 1);
+
+        dmGameObject::HCollection collection = dmGameObject::GetCollection(CheckGoInstance(L));
+        void* comp = 0x0;
+        void* comp_world = 0x0;
+        GetCollisionObject(L, 1, collection, &comp, &comp_world);
+        dmhash_t group_id = dmScript::CheckHashOrString(L, 2);
+        
+        bool boolvalue = dmGameSystem::GetCollisionMaskBit(comp_world, comp, group_id);
```

Comment:
Simply returning a boolean value doesn't catch the case where the `group_id` doesn't exist.

Reply:
There is a warning already written to log (in GetGroupBitIndex()) . 

What's the proper way to handle this ?